sdk/go/common/tokens/project.go

```diff
@@ -33,6 +37,9 @@ func ValidateProjectName(s string) error {
 		return errors.New("project names are limited to 100 characters")
 	case !IsName(s):
 		return errors.New("project names may only contain alphanumerics, hyphens, underscores, and periods")
+	case strings.ToLower(s) == "pulumi" || strings.HasPrefix(strings.ToLower(s), "pulumi."):
+		return errors.New("project name must not be `pulumi` and must not start with the prefix `pulumi.` " +
+			"to avoid collision with standard libraries")
```

Comment:
I'm totally blanking on "avoid collision with standard libraries". Can you explain?